### PR TITLE
Implement dynamic timestamp types in Message class

### PR DESCRIPTION
**Pull request by massooti**
Aligning with confluent-kafka behavior,
Implemented broker receive time simulation for
Updated timestamp() method to return a tuple[int, int] as per confluent-kafka specs.

### CODE REVIEW

**Comment by alm0ra** [comment]
Hey @massooti , thanks for this PR , it seems there are some typing issues.

**Review by PeterJCLaw — COMMENTED** [review]
Thanks for this PR, however there seems to be a lot of extraneous changes here which aren't related to the feature fix. In particular, I would prefer that the mypy configuration changes be undone as the previous configuration was deliberate to avoid masking errors. (I would also encourage undoing the various layout/whitespace changes which seem to have crept into quite a few of the files but have no bearing on the core change).

As a slightly meta point, this project uses merge commits to combine branches, so if you wanted to rebase/squash some of the history once the extraneous changes have been removed in order to prepare the commits for review that would be fine.

**Comment by alm0ra** [comment]
Hey @massooti,

I opened a PR #151 to demonstrate that our GitHub Action with MyPy is functioning correctly.

Regarding your PR:

Please disable your IDE's reformatter and revert any unrelated changes.
Kindly provide test cases that cover the affected code.
Thanks!

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.mypy]
 warn_unused_ignores = true
-
+ignore_missing_imports = true
```
[the remark this answers is not in the document]
Comment (massooti) [reply pyproject.toml]: Thanks for the feedback! I've addressed all the concerns you mentioned. I reverted the global ignore and added inline ignores where necessary, focusing on specific cases. I also avoided changing any third-party linting configurations.

**Comment by massooti** [comment]
finally resolved poetry conflicts. its ready to merge

**Review by alm0ra — APPROVED** [review]

**Pull request merged**